cpp/CMakeLists.txt

```diff
@@ -339,16 +355,30 @@ message(STATUS "CUDF_TEST_LIST set to: ${CUDF_TEST_LIST}")
 # - add google benchmark --------------------------------------------------------------------------
 
 if(BUILD_BENCHMARKS)
+    find_package(benchmark)
 
-  include(ConfigureGoogleBenchmark)
+    if(benchmark_ROOT)
+        message(STATUS "Google C++ Benchmarking Framework (Google Benchmark) found in ${benchmark_ROOT}")
+        add_subdirectory(${CMAKE_SOURCE_DIR}/benchmarks)
+    else()
+        message(AUTHOR_WARNING "Google C++ Benchmarking Framework (Google Benchmark) not found, it will be built from source.")
+        include(ConfigureGoogleBenchmark)
+        add_subdirectory(${CMAKE_SOURCE_DIR}/benchmarks)
+        set(benchmark_ROOT "${CMAKE_BINARY_DIR}/googlebenchmark")
+    endif(benchmark_ROOT)
 
-  if(GBENCH_FOUND)
-    message(STATUS "Google C++ Benchmarking Framework (Google Benchmark) found in ${GBENCH_ROOT}")
-    include_directories(${GBENCH_INCLUDE_DIR})
-    add_subdirectory(${CMAKE_SOURCE_DIR}/benchmarks)
-  else()
-    message(AUTHOR_WARNING "Google C++ Benchmarking Framework (Google Benchmark) not found: automated tests are disabled.")
-  endif(GBENCH_FOUND)
+    set(benchmark_INCLUDE_DIR "${benchmark_ROOT}/install/include")
+    set(benchmark_LIBRARY_DIR "${benchmark_ROOT}/install/lib" "${benchmark_ROOT}/install/lib64")
+    include_directories(${benchmark_INCLUDE_DIR})
+
+    message(STATUS "benchmark_ROOT: " ${benchmark_ROOT})
+
+    if(benchmark_FOUND)
+        message(STATUS "Google C++ Benchmarking Framework (Google Benchmark) found in ${benchmark_ROOT}")
+    else()
+        message(AUTHOR_WARNING "Google C++ Benchmarking Framework (Google Benchmark) not found: automated tests are disabled.")
+        set(BUILD_BENCHMARKS OFF)
+    endif()
```

Comment:
I think this chunk can be removed now? We either find an installed version or build it from source ourselves.

Reply:
I think it still serves a purpose. The first checks for the root even if not found. In my tests I've found the if I do `benchmark_FOUND` only or `benchmark_ROOT` and there is no `conda` package or other linked install and it does the install, then the installation can fail to link for the subsequent processes without explicitly setting and doing a check. Also the bottom check if the install failed ensures that `BUILD_BENCHMARKS` is switched to `OFF` if a failure has occurred preventing that variable from being misused in further compilation. 

I get the idea of only using `benchmark_FOUND` and calling it good, but in my tests I've found that not to perform consistently.